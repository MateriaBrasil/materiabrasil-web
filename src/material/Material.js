import { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import renderInfo from '../renderInfo'

import * as actions from './actions'

class Material extends Component {
  async componentWillMount() {
    const { actions, match } = this.props
    const { params } = match
    const { id } = params
    const { info } = actions
    await info(id)
  }

  render() {
    return renderInfo(this.props)
  }
}

export default connect(
  state => ({ ...state.material }),
  dispatch => ({ actions: bindActionCreators(actions, dispatch) }),
)(Material)
