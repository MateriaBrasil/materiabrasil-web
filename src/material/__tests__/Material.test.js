import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import createStore from '../../store/createStore'

import Material from '../Material'

jest.mock('../actions', () => ({ info: () => ({ type: 'foo' }) }))

it('renders correctly', () => {
  const state = { material: {} }
  const reducer = () => state
  const store = createStore(reducer)
  const match = { params: { id: 1 } }

  const tree = renderer
    .create(
      <Provider store={store}>
        <Material match={match} render={props => <div {...props} />} />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly while starting up', () => {
  const state = { material: { startingUp: true } }
  const reducer = () => state
  const store = createStore(reducer)
  const match = { params: { id: 1 } }

  const tree = renderer
    .create(
      <Provider store={store}>
        <Material
          match={match}
          {...state}
          render={props => <div {...props} />}
        />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly while loading', () => {
  const state = { material: { requestingInfo: true } }
  const reducer = () => state
  const store = createStore(reducer)
  const match = { params: { id: 1 } }

  const tree = renderer
    .create(
      <Provider store={store}>
        <Material
          match={match}
          {...state}
          render={props => <div {...props} />}
        />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with errors', () => {
  const state = { material: { infoError: 'foo' } }
  const reducer = () => state
  const store = createStore(reducer)
  const match = { params: { id: 1 } }

  const tree = renderer
    .create(
      <Provider store={store}>
        <Material match={match} {...state} />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
