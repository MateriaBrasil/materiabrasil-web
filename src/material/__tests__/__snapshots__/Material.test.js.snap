// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Info
</div>
`;

exports[`renders correctly while loading 1`] = `
<div
  className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar1Indeterminate-12"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar2Indeterminate-13"
    style={Object {}}
  />
</div>
`;

exports[`renders correctly while starting up 1`] = `
<div
  className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar1Indeterminate-12"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar2Indeterminate-13"
    style={Object {}}
  />
</div>
`;

exports[`renders correctly whith errors 1`] = `
<div
  style={
    Object {
      "padding": 20,
      "textAlign": "center",
    }
  }
>
  <h1
    className="MuiTypography-root-17 MuiTypography-headline-22 MuiTypography-colorError-40"
  >
    foo
  </h1>
</div>
`;
